### PR TITLE
Add Array#pluck method as alias to array.map { |h| h[:a] }

### PR DESCRIPTION
**Pull request by divineforest**
Solves common pattern:

```
x = [{ a: 1, b: 2 }, { a: 10, b: 20 }]
x.map { |h| h[:a] }
```

This method allows to achieve the same with:

```
x.pluck(:a)
```

### CODE REVIEW

File: activesupport/lib/active_support/core_ext/array/pluck.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  #
+  #   x = [{ a: 1, b: 2 }, { a: 10, b: 20 }]
+  #
+  #   instead of
+  #   x.map { |h| h[:a] }
+  #
+  #   x.pluck(:a) => [1, 10]
+  #   x.pluck(:a, :b) => [[1, 2], [10, 20]]
+  def pluck(*keys)
+    if keys.size > 1
```
Comment (killthekitten) [inline activesupport/lib/active_support/core_ext/array/pluck.rb]: `map` with `slice` could look more elegant and compact, are there any problems with using it?

Comment (killthekitten) [inline activesupport/lib/active_support/core_ext/array/pluck.rb]: in fact, I was meaning `values_at`, sorry about that

**Comment by seuros** [comment]
This is a new addition, a change-log is required.

File: activesupport/lib/active_support/core_ext/array/pluck.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class Array
+  # Extracts values from array of hashes.
```
Comment (killthekitten) [inline activesupport/lib/active_support/core_ext/array/pluck.rb]: You should mention here, that method is sensitive to string/symbol arguments.

**Comment by divineforest** [comment]
@seuros done

**Comment by sgrif** [comment]
We don't generally accept new methods to Active Support, unless they demonstrate a significant value within the framework's code itself.

**Comment by rafaelfranca** [comment]
Yes, even more a core extension that expect the object contains a specific type of object. What would happen if people call `pluck` in an Array of Integer?

**Comment by divineforest** [comment]
@sgrif  @rafaelfranca I agree that it is not ideal for now. May it should be renamed to `map_hash` and check for objects type. But this patterns is so often that I thought it would be convenient to extract it to method. May there is a better place or form of doing this?

**Comment by sgrif** [comment]
I don't think this is a common pattern at all. In fact, I'd say it's probably a code smell that you're missing an object.

> May there is a better place or form of doing this?

If you think this is something that is needed, I'd recommend submitting a patch to the Ruby language.

**Comment by rafaelfranca** [comment]
Yes. Agree

**Comment by rafaelfranca** [comment]
Thank you for the pull requests

**Pull request closed without merging**